test/favorite_locations_test.dart

```diff
@@ -0,0 +1,16 @@
+import 'package:one_app/utils/data_service.dart';
+import 'package:one_app/utils/favorite_locations.dart';
+import 'package:one_app/utils/location_data.dart';
+import 'package:test/test.dart';
+
+void main() {
+  test('element should be added to set', () {
+    final favoriteLocations = FavoriteLocations();
+    final dataService = DataService();
+    final locationData = LocationData('New York', dataService);
+
+    favoriteLocations.addLocation(locationData);
+
+    expect(favoriteLocations.currentLocations.contains(locationData), true);
```

Comment:
this is perfectly fine as-is. Some people do prefer using a matcher here, though, since that improves readability slightly:

```suggestion
    expect(favoriteLocations.currentLocations, contains(locationData));
```